### PR TITLE
update mujoco and tensorflow 1.14/2.0 api calls in efficient-hrl

### PR DESCRIPTION
**Pull request by krishpop**
Fixes issue addressed in #5321, updating mujoco calls to `self.data` and TensorFlow calls to `tf.CriticalSection` to reflect 1.14/2.0 standard.

### CODE REVIEW

File: research/efficient-hrl/agents/circular_buffer.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def __init__(self, buffer_size=1000, scope='replay_buffer'):
     self._tensors = collections.OrderedDict()
     with tf.variable_scope(self._scope):
       self._num_adds = tf.Variable(0, dtype=tf.int64, name='num_adds')
-    self._num_adds_cs = tf.contrib.framework.CriticalSection(name='num_adds')
+    self._num_adds_cs = tf.CriticalSection(name='num_adds')
```
Comment (ofirnachum) [inline research/efficient-hrl/agents/circular_buffer.py]: thanks for this

File: research/efficient-hrl/environments/ant.py [hunk 2]
```diff
@@ -75,13 +75,13 @@ def _get_obs(self):
     # No cfrc observation
     if self._expose_all_qpos:
       obs = np.concatenate([
-          self.physics.data.qpos.flat[:15],  # Ensures only ant obs.
-          self.physics.data.qvel.flat[:14],
+          self.data.qpos.flat[:15],  # Ensures only ant obs.
```
Comment (ofirnachum) [inline research/efficient-hrl/environments/ant.py]: Is there a way to get this to work regardless of mujoco 2.0 or <2.0?  Or at least make a property method that either returns self.data or self.physics.data depending on which mujoco version?

File: research/efficient-hrl/environments/ant.py [hunk 3]
```diff
@@ -117,18 +117,18 @@ def viewer_setup(self):
 
   def get_ori(self):
     ori = [0, 1, 0, 0]
-    rot = self.model.data.qpos[self.__class__.ORI_IND:self.__class__.ORI_IND + 4]  # take the quaternion
+    rot = self.data.qpos[self.__class__.ORI_IND:self.__class__.ORI_IND + 4]  # take the quaternion
```
Comment (ofirnachum) [inline research/efficient-hrl/environments/ant.py]: same comment as above

File: research/efficient-hrl/environments/maze_env.py [hunk 4]
```diff
@@ -228,7 +228,7 @@ def __init__(
         raise Exception("Every geom of the torso must have a name "
                         "defined")
 
-    _, file_path = tempfile.mkstemp(text=True)
+    _, file_path = tempfile.mkstemp(text=True, suffix='.xml')
```
Comment (ofirnachum) [inline research/efficient-hrl/environments/maze_env.py]: thanks

File: research/efficient-hrl/environments/ant.py (hunk 2 above)
Comment (krishpop) [reply research/efficient-hrl/environments/ant.py]: I see, sorry for the late response on this, I got a bit busy. I modified the ant and point environments to revert back to using self.physics, and then used the mujoco version number to get the right attribute (self.model or self.sim) based on the mujoco version. Let me know if this is more or less what you meant by this.

File: research/efficient-hrl/environments/ant.py [hunk 5]
```diff
@@ -117,13 +124,13 @@ def viewer_setup(self):
...
     ori = [0, 1, 0, 0]
-    rot = self.model.data.qpos[self.__class__.ORI_IND:self.__class__.ORI_IND + 4]  # take the quaternion
+    rot = self.physics.data.qpos[self.__class__.ORI_IND:self.__class__.ORI_IND + 4]  # take the quaternion
     ori = q_mult(q_mult(rot, ori), q_inv(rot))[1:3]  # project onto x-y plane
     ori = math.atan2(ori[1], ori[0])
     return ori
 
   def set_xy(self, xy):
-    qpos = np.copy(self.physics.data.qpos)
+    qpos = np.copy(self.data.qpos)
```
Comment (ofirnachum) [inline research/efficient-hrl/environments/ant.py]: Should this have been changed?  Or left as self.physics.data.qpos?

**Pull request merged**